engine/resource/src/resource.cpp

```diff
@@ -856,6 +859,8 @@ Result InsertResource(HFactory factory, const char* path, uint64_t canonical_pat
         factory->m_ResourceHashToFilename->Put(canonical_path_hash, strdup(canonical_path));
     }
 
+    descriptor->m_Version = factory->m_Version++;
```

Comment:
Shouldn't we check for overflow and start from the beginning to avoid any undefined issues when value will be bigger than 65535?